<?xml version="1.0"?>

<launch>

    <arg name="markerSize"        default="0.5"/>    <!-- in m -->
    <arg name="dct_normalization" default="False" />
    <arg name="dct_filter_size"   default="2" />
    <arg name="camera_frame"      default="camera_rgb_optical_frame"/>
    <arg name="marker_frame"     default="aruco_frame" />



    <node pkg="aruco_ros" type="double" name="aruco_simple" output="screen">>    
        <remap from="/camera_info" to="/camera/rgb/camera_info" />
        <remap from="/image" to="/camera/rgb/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size" value="$(arg markerSize)"/>

        <param name="normalizeImage" value="$(arg dct_normalization)"/>
        <param name="dct_components_to_remove" value="$(arg dct_filter_size)"/>
        <param name="parent_name" value="$(arg camera_frame)"/>
        <param name="child_name"  value="$(arg marker_frame)" />


    </node>

</launch>

